music21/interval.py

```diff
@@ -318,10 +318,7 @@ def _extractPitch(nOrP: Union['music21.note.Note', 'music21.pitch.Pitch']):
     True
 
     '''
-    if hasattr(nOrP, 'pitch'):
-        return nOrP.pitch
-    else:
-        return nOrP
+    return getattr(nOrP, 'pitch', nOrP)
```

Comment:
Mypy really hates `hasattr`. But if this feels illegible, do say so.

Reply:
yes, that feels illegible, sorry to say.  I'd rather disable mypy for the function than have something that isn't clear what it does.  I still use music21 code as a teaching tool, and I don't like to encourage my students to write something like this.

The approach in https://github.com/python/mypy/issues/1424#issuecomment-894179881  is acceptable if commented.  But I really don't understand why getattr is acceptable but hasattr is not.